astropy/io/fits/hdu/compressed/_tiled_compression.py

```diff
@@ -359,7 +360,7 @@ def decompress_image_data_section(
     else:
         zzero_column = None
 
-    zblank_header = compressed_header.get("ZBLANK", None)
+    zblank_header = compressed_header.get("ZBLANK", image_header.get("BLANK", None))
```

Comment:
Just to check, is BLANK not defined also in the compressed header if we are decompressing data? Or is the idea here to allow a user to change BLANK after opening a file but before decompressing it?

Reply:
It doesn't work for the last test case: zblank column without z/blank in the header. In that case we use the blank value that is set below in `header.py` and is not in the compressed header.
I tried to remove that line, we could pass the blank value by another mean, but in the end it was not looking much better so I kept that.